### PR TITLE
[Flare] Adds useListener implementation to ReactDebugHooks

### PR DESCRIPTION
**Pull request by trueadm**
This adds the implementation of `useListener` to `ReactDebugHooks`.

### CODE REVIEW

File: packages/react-debug-tools/src/ReactDebugHooks.js [hunk 1]
```diff
@@ -215,8 +221,16 @@ function useMemo<T>(
   return value;
 }
 
-function useListener() {
-  throw new Error('TODO: not yet implemented');
+function useListener(
+  responder: ReactEventResponder<any, any>,
+  hookProps: ?Object,
+): void {
+  const listenerProps = hookProps || emptyObject;
```
Comment (bvaughn) [inline packages/react-debug-tools/src/ReactDebugHooks.js]: Any reason we couldn't just use `props: null` in the `hookLog` ?

File: packages/react-debug-tools/src/ReactDebugHooks.js [hunk 2]
```diff
@@ -215,8 +221,17 @@ function useMemo<T>(
...
+  // Don't put the actual event responder object in, just its displayName
+  const value = {
+    responder: responder.displayName,
```
Comment (bvaughn) [inline packages/react-debug-tools/src/ReactDebugHooks.js]: Should we set a default/fallback name (we do elsewhere for event components)

Comment (trueadm) [reply packages/react-debug-tools/src/ReactDebugHooks.js]: Event responders should always have a mandatory part of the `unstable_createResponder` API. I can add that though!

Comment (bvaughn) [reply packages/react-debug-tools/src/ReactDebugHooks.js]: Ah! Okay well if we're _guaranteed_ to have a name then no need. I just wasn't sure, based on the precedent in the renderer interface. 😄

Comment (trueadm) [reply packages/react-debug-tools/src/ReactDebugHooks.js]: I added it anyway, to be safe!

**Pull request merged**